exercises/robot-simulator/src/test/java/RobotTest.java

```diff
@@ -28,184 +28,163 @@ public void testNegativePositionsAreAllowed() {
     }
 
     @Ignore("Remove to run test")
     @Test
-    public void testAdvancingDoesNotChangeOrientation() {
+    public void increasesTheYCoordinateOneWhenFacingNorth() {
         Orientation initialOrientation = Orientation.NORTH;
         Robot robot = new Robot(new GridPosition(0, 0), initialOrientation);
 
         robot.advance();
 
-        assertEquals(robot.getOrientation(), initialOrientation);
-    }
-
-    @Ignore("Remove to run test")
-    @Test
-    public void testAdvancingWhenFacingNorthIncreasesYCoordinateByOne() {
-        Robot robot = new Robot(new GridPosition(0, 0), Orientation.NORTH);
-
-        robot.advance();
-
         GridPosition expectedGridPosition = new GridPosition(0, 1);
-
         assertEquals(robot.getGridPosition(), expectedGridPosition);
+        assertEquals(robot.getOrientation(), initialOrientation);
     }
 
     @Ignore("Remove to run test")
     @Test
-    public void testAdvancingWhenFacingSouthDecreasesYCoordinateByOne() {
-        Robot robot = new Robot(new GridPosition(0, 0), Orientation.SOUTH);
+    public void decreasesTheYCoordinateByOneWhenFacingSouth() {
+        Orientation initialOrientation = Orientation.SOUTH;
+        Robot robot = new Robot(new GridPosition(0, 0), initialOrientation);
 
         robot.advance();
 
         GridPosition expectedGridPosition = new GridPosition(0, -1);
-
         assertEquals(robot.getGridPosition(), expectedGridPosition);
+        assertEquals(robot.getOrientation(), initialOrientation);
     }
 
     @Ignore("Remove to run test")
     @Test
-    public void testAdvancingWhenFacingEastIncreasesXCoordinateByOne() {
-        Robot robot = new Robot(new GridPosition(0, 0), Orientation.EAST);
+    public void increasesTheXCoordinateByOneWhenFacingEast() {
+        Orientation initialOrientation = Orientation.EAST;
+        Robot robot = new Robot(new GridPosition(0, 0), initialOrientation);
 
         robot.advance();
 
         GridPosition expectedGridPosition = new GridPosition(1, 0);
-
         assertEquals(robot.getGridPosition(), expectedGridPosition);
+        assertEquals(robot.getOrientation(), initialOrientation);
     }
 
     @Ignore("Remove to run test")
     @Test
-    public void testAdvancingWhenFacingWestDecreasesXCoordinateByOne() {
-        Robot robot = new Robot(new GridPosition(0, 0), Orientation.WEST);
+    public void decreasesTheXCoordinateByOneWhenFacingWest() {
+        Orientation initialOrientation = Orientation.WEST;
+        Robot robot = new Robot(new GridPosition(0, 0), initialOrientation);
 
         robot.advance();
 
         GridPosition expectedGridPosition = new GridPosition(-1, 0);
-
         assertEquals(robot.getGridPosition(), expectedGridPosition);
+        assertEquals(robot.getOrientation(), initialOrientation);
     }
 
     @Ignore("Remove to run test")
     @Test
-    public void testInstructionsToMoveWestAndNorth() {
+    public void instructionsToMoveWestAndNorth() {
```

Comment:
I think there should be a test before this "instructions to move east and north" in the canonical data? 🙂

Reply:
I used the description field of the canonical data as test name, link to case
https://github.com/exercism/problem-specifications/blob/3672d599915d5ba39bbe20b3e04859f1d852a51e/exercises/robot-simulator/canonical-data.json#L357

Reply:
there are two cases with same description
https://github.com/exercism/problem-specifications/blob/3672d599915d5ba39bbe20b3e04859f1d852a51e/exercises/robot-simulator/canonical-data.json#L300
and
https://github.com/exercism/problem-specifications/blob/3672d599915d5ba39bbe20b3e04859f1d852a51e/exercises/robot-simulator/canonical-data.json#L357

What I have to do ? I cannot have two test methods with the same name

Reply:
Hi @mirkoperillo - the test names do not have to be the exact same as the test description. Each test normally serves a distinct purpose. Can you spot a difference between the two and name them accordingly? :slightly_smiling_face:

Reply:
pushed my proposal